src/main/java/net/mcreator/ui/views/editor/image/tool/tools/FlipHorizontalTool.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2025, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.views.editor.image.tool.tools;
+
+import net.mcreator.ui.component.zoompane.ZoomedMouseEvent;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.init.UIRES;
+import net.mcreator.ui.views.editor.image.canvas.Canvas;
+import net.mcreator.ui.views.editor.image.canvas.Selection;
+import net.mcreator.ui.views.editor.image.tool.component.ColorSelector;
+import net.mcreator.ui.views.editor.image.versioning.VersionManager;
+
+import java.awt.*;
+import java.awt.event.MouseEvent;
+import java.awt.image.BufferedImage;
+
+public class FlipHorizontalTool extends AbstractModificationTool {
+	public FlipHorizontalTool(Canvas canvas, ColorSelector colorSelector, VersionManager versionManager) {
+		super(L10N.t("dialog.image_maker.tools.types.flip_horizontal"),
+				L10N.t("dialog.image_maker.tools.types.flip_horizontal_description"),
+				UIRES.get("img_editor.fliphorizontal"), canvas, colorSelector, versionManager);
+		noSettings(true);
+	}
+
+	@Override public boolean process(ZoomedMouseEvent e) {
+		flipImage(layer.getRaster());
+		return true;
+	}
+
+	@Override public void mouseDragged(MouseEvent e) {
+	}
+
+	public void flipImage(BufferedImage image) {

```

Comment:
Maybe instead of manually processing pixels, you could use negative size to flip it:

```
g2d.setComposite(AlphaComposite.Clear);
g2d.fillRect(minX, minY, width, height);
g2d.setComposite(AlphaComposite.SrcOver);
g2d.drawImage(image.getSubimage(minX, minY, width, height), minX + width, minY, -width, height, null);
```

(untested code, just a concept)

It may also make sense to put the whole processing in the process method itself looking at how other tools are coded